src/contract/event.rs

```diff
@@ -329,60 +297,56 @@ impl<T: Transport, E: ParseLog> AllEventsBuilder<T, E> {
     /// For regular events, this corresponds to the event signature. For
     /// anonymous events, this is the first indexed property.
     pub fn topic0(mut self, topic: Topic<H256>) -> Self {
-        self.topics.topic0 = topic;
+        self.filter = self.filter.topic0(topic);
         self
     }
 
     /// Adds a filter for the second indexed topic.
     pub fn topic1(mut self, topic: Topic<H256>) -> Self {
-        self.topics.topic1 = topic;
+        self.filter = self.filter.topic1(topic);
         self
     }
 
     /// Adds a filter for the third indexed topic.
     pub fn topic2(mut self, topic: Topic<H256>) -> Self {
-        self.topics.topic2 = topic;
+        self.filter = self.filter.topic2(topic);
         self
     }
 
     /// Adds a filter for the third indexed topic.
     pub fn topic3(mut self, topic: Topic<H256>) -> Self {
-        self.topics.topic2 = topic;
-        self
-    }
-
-    /// The polling interval. This is used as the interval between consecutive
-    /// `eth_getFilterChanges` calls to get filter updates.
-    pub fn poll_interval(mut self, value: Duration) -> Self {
-        self.poll_interval = Some(value);
+        self.filter = self.filter.topic3(topic);
         self
     }
 
     /// The page size in blocks to use when doing a paginated query on past
     /// events.
     pub fn block_page_size(mut self, value: u64) -> Self {
-        self.block_page_size = Some(value);
+        self.filter = self.filter.block_page_size(value);
```

Comment:
This seems incredibly weird to me...

Like why not just `self.filter.block_page_size = value`?

Is this some sort of meta recursion?

Reply:
This is because `LogFilterBuilder::block_page_size` takes in a `LogFilterBuilder` as `self` and returns an updated `LogFilterBuilder`. Also, `self.filter.block_page_size(value)` does some value checking (for example panicking if `value == 0`).